kie-internal/src/main/java/org/kie/internal/io/ResourceTypePackage.java

```diff
@@ -15,8 +15,20 @@
 
 package org.kie.internal.io;
 
+import org.kie.api.io.Resource;
 import org.kie.api.io.ResourceType;
 
 public interface ResourceTypePackage {
     ResourceType getResourceType();
+
+    /**
+     * Remove artifacts inside this ResourceTypePackage which belong to the resource passed as parameter.
+     * Concrete implementation of this interface shall extend this method in order to properly support incremental KieContainer updates.
+     * 
+     * @param resource
+     * @return true if this ResourceTypePackage mutated as part of this method invocation.
+     */
+    default boolean removeFromResource(Resource resource) {
```

Comment:
@tarilabs the name of the method seems confusing to me. Maybe rename it to "removeResource()"?